src/main/java/org/websoso/WSSServer/repository/NovelCustomRepositoryImpl.java

```diff
@@ -0,0 +1,74 @@
+package org.websoso.WSSServer.repository;
+
+import static org.websoso.WSSServer.domain.QNovel.novel;
+import static org.websoso.WSSServer.domain.QUserNovel.userNovel;
+
+import com.querydsl.core.types.dsl.BooleanExpression;
+import com.querydsl.core.types.dsl.Expressions;
+import com.querydsl.core.types.dsl.NumberTemplate;
+import com.querydsl.core.types.dsl.StringTemplate;
+import com.querydsl.jpa.impl.JPAQueryFactory;
+import java.util.Collections;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Repository;
+import org.websoso.WSSServer.domain.Novel;
+
+@Repository
+@RequiredArgsConstructor
+public class NovelCustomRepositoryImpl implements NovelCustomRepository {
+
+    private final JPAQueryFactory jpaQueryFactory;
+
+    @Override
+    public Page<Novel> findSearchedNovels(Pageable pageable, String query) {
+
+        if (query.isEmpty() || query.isBlank()) {
+            return new PageImpl<>(Collections.emptyList(), pageable, 0);
+        }
+
+        String searchQuery = query.replaceAll("\\s+", "");
+
+        StringTemplate cleanTitle = Expressions.stringTemplate(
+                "REPLACE(REPLACE({0}, ' ', ''), CHAR(9), '')",
+                novel.title
+        );
+        StringTemplate cleanAuthor = Expressions.stringTemplate(
+                "REPLACE(REPLACE({0}, ' ', ''), CHAR(9), '')",
+                novel.author
+        );
+
+        NumberTemplate<Long> popularity = Expressions.numberTemplate(Long.class,
+                "(SELECT COUNT(un) FROM UserNovel un WHERE un.novel = {0} AND (un.isInterest = true OR un.status <> 'QUIT'))",
+                novel);
```

Comment:
p2;
이렇게 하드 코딩으로 쿼리를 작성하는 것은 query dsl을 사용하지 않아도 가능할 것 같은데요,
JPAQueryFactory로도 가독성 좋고 하드 코딩하지 않게 만들 수 있지 않을까요!?

Reply:
```java
private NumberTemplate<Long> getPopularity(QNovel novel) {
    return Expressions.numberTemplate(
            Long.class,
            "COUNT(userNovel.userNovelId)",
            jpaQueryFactory.selectFrom(userNovel)
                    .where(userNovel.novel.eq(novel)
                            .and(userNovel.status.ne(QUIT)
                                    .or(userNovel.isInterest.isTrue())))
    );
}
```
다음과 같이 수정했습니다!

Reply:
CaseBuilder로 구현해보시는 건 어떨까요!?

```java
private NumberExpression<Long> getPopularity(QNovel novel) {
    return new CaseBuilder()
            .when(userNovel.status.in(WATCHING, WATCHED)
                    .or(userNovel.isInterest.isTrue()))
            .then(1L)
            .otherwise(0L)
            .sum();
}
```

Reply:
이게 훨씬 더 깔끔하네요! 반영했습니다~